accessible/fieldview.component.js

```diff
@@ -0,0 +1,146 @@
+            [attr.aria-level]="level+1" aria-selected=false>
+          <button [id]="idMap[optionValue + 'Button']" (click)="handleDropdownChange(field,optionValue)">
+            {{optionText[optionValue]}}
+          </button>
+        </li>
+      </ol>
+    </li>
+    <li [id]="idMap['listItem']" role="treeitem" 
+        *ngIf="isCheckbox(field)" [attr.aria-level]="level" 
+        aria-selected=false>
+      // Checkboxes not currently supported.
+    </li>
+    <li [id]="idMap['listItem']" role="treeitem" 
+        [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-argument-text', idMap['label'])" 
+        *ngIf="isTextField(field) && hasVisibleText(field)" 
+        [attr.aria-level]="level" aria-selected=false>
+      <label [id]="idMap['label']">
+        {{field.getText()}}
+      </label>
+    </li>
+    `,
+    inputs: ['field', 'level', 'index', 'parentId'],
+    providers: [blocklyApp.TreeService, blocklyApp.UtilsService],
+  })
+  .Class({
+    constructor: [blocklyApp.TreeService, blocklyApp.UtilsService, 
+        function(_treeService, _utilsService) {
+      this.optionText = {
+        keys: []
+      };
+      this.text = '';
+      this.treeService = _treeService;
+      this.utilsService = _utilsService;
+      this.stringMap = {
+        'CURRENT_ARGUMENT_VALUE': Blockly.Msg.CURRENT_ARGUMENT_VALUE,
+      };
+    }],
+    ngOnInit: function() {
+      var elementsNeedingIds = this.generateElementNames(this.field);
+      this.idMap = this.utilsService.generateIds(elementsNeedingIds);
+    },
+    generateElementNames: function(field) {
+      var elementNames = ['listItem'];
+      switch(true) {
+        case this.isTextInput(field):
+          elementNames.push('input');
+          break;
+        case this.isDropdown(field):
+          elementNames.push('label');
+          var keys = this.getOptions(field);
+          for (var i = 0; i < keys.length; i++){
+            elementNames.push(keys[i]);
+            elementNames.push(keys[i] + 'Button');
+          }
+          break;
+        case this.isTextField(field) && this.hasVisibleText(field):
+          elementNames.push('label');
+          break;
+        default:
+          break;
+      }
+      return elementNames;
+    },
+    isTextInput: function(field) {
+      return field instanceof Blockly.FieldTextInput;
+    },
+    isDropdown: function(field) {
+      return field instanceof Blockly.FieldDropdown;
+    },
+    isCheckbox: function(field) {
+      return field instanceof Blockly.FieldCheckbox;
+    },
+    isTextField: function(field) {
+      return !(field instanceof Blockly.FieldTextInput) &&
+          !(field instanceof Blockly.FieldDropdown) &&
+          !(field instanceof Blockly.FieldCheckbox);
+    },
+    hasVisibleText: function(field) {
+      var text = field.getText().trim();
+      return !!text;
+    },
+    getOptions: function(field) {
```

Comment:
This seems to be getting the options for a particular field. However, once it's called once, then this.optionText.keys will have a nonzero length, and the options won't be recomputed again, even if they're different for different fields.

Is this intentional?